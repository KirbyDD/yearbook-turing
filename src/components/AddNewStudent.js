import React, { Component } from 'react';

class AddNewStudent extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      quote: '',
      superlative: '',
      form: false     
    }
  }

  handleChange = (event) => {
    this.setState({[event.target.name]: event.target.value})
  }

  createForm = () => {
    this.setState({ form: true })
  }

  makeStudent = () => {
    let newStudent = {
      id: Date.now(),
      name: this.state.name,
      quote: this.state.quote,
      superlative: this.state.superlative,
      photo: 'https://placekitten.com/200/300',
    }
    this.props.addStudent(newStudent);
    this.resetForm()
  }

  resetForm = () => {
    this.setState({
      name: '',
      quote: '',
      superlative: '',
      form: false
    })
  }

  render(){

    return (
      <section>
        <button onClick={event => this.createForm() }>Add Student!</button>
        {this.state.form && 
          <form onChange={event => this.handleChange(event)}>
            <input type='text' placeholder='name' value={this.state.name} name='name'/>
            <input type='text' placeholder='quote' value={this.state.quote} name='quote'/>
            <input type='text' placeholder='superlative' value={this.state.superlative} name='superlative'/>
            <button onClick={event => this.makeStudent()}>Submit</button>
          </form>
        }
      </section>
    )
  }
}

export default AddNewStudent;