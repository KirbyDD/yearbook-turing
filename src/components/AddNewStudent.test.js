import React from 'react';
import AddNewStudent from './AddNewStudent';
import { shallow } from 'enzyme'

describe('AddNewStudent', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<AddNewStudent staff={[]} students={[]}/>)
    expect(wrapper).toMatchSnapshot();
  })
})