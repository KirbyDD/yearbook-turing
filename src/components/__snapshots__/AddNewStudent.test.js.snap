// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewStudent should match snapshot 1`] = `ShallowWrapper {}`;
